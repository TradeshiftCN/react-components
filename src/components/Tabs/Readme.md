### Basic Usage

```jsx
const { TabPane } = Tabs;
const onChange = key => {
	console.log(key);
};

<Tabs defaultActiveKey="2" onChange={onChange}>
	<TabPane tab="One" key="1">
		first
	</TabPane>
	<TabPane tab="Two" key="2">
		second
	</TabPane>
	<TabPane tab="Three" key="3">
		third
	</TabPane>
</Tabs>;
```

For more options, please view [rc-tabs](https://github.com/react-component/tabs)