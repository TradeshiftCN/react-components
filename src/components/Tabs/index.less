@import '../../style/variables/index';

.ts-rc-tabs {
	font-size: @ts-fontsize;
	&-bar {
		position: relative;
		line-height: @ts-unit-double;
		color: @ts-color-slate-lightest;
		font-weight: @ts-fontweight-semibold;
		user-select: none;
	}
	&-tab {
		position: relative;
		display: inline-block;
		padding: 0 @ts-unit-half;
		transition: @ts-transition-now;
		cursor: pointer;
		&-active,
		&:hover {
			color: @ts-color-black;
		}
	}
	&-ink-bar {
		height: 2px;
		position: absolute;
		left: 0;
		bottom: 0;
		transition: 50ms;
		background-color: @ts-color-black;
	}
}
