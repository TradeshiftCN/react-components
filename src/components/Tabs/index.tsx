import React from 'react';
import PropTypes from 'prop-types';
import RcTabs, { TabPane } from 'rc-tabs';
import TabContent from 'rc-tabs/lib/TabContent';
import InkTabBar from 'rc-tabs/lib/InkTabBar';

type TabsProps = {
	activeKey?: string;
	defaultActiveKey?: string;
	renderTabBar?(): React.ReactNode;
	renderTabContent?(): React.ReactNode;
	onChange?(key: string): void;
	destroyInactiveTabPane?: boolean;
};

const prefixCls = 'ts-rc-tabs';

function Tabs(props: TabsProps) {
	return <RcTabs prefixCls={prefixCls} {...props} />;
}

Tabs.propTypes = {
	defaultActiveKey: PropTypes.string,
	/** ()=> React.ReactNode */
	renderTabBar: PropTypes.func,
	/** ()=> React.ReactNode */
	renderTabContent: PropTypes.func,
	/** (key: string)=> void */
	onChange: PropTypes.func,
	/** whether destroy inactive tabpane when change tab */
	destroyInactiveTabPane: PropTypes.bool
};

Tabs.defaultProps = {
	destroyInactiveTabPane: true,
	renderTabBar: () => <InkTabBar />,
	renderTabContent: () => <TabContent animated={false} />
};

Tabs.TabPane = TabPane;

export default Tabs;
