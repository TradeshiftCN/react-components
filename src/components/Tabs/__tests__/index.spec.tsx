import React from 'react';
import { shallow } from 'enzyme';
import Tabs from '..';

describe('Tabs', () => {
	it('basic', () => {
		const wrapper = shallow(<Tabs />);

		expect(wrapper).toMatchSnapshot();
	});
});
