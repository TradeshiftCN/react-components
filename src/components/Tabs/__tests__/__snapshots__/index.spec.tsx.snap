// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs basic 1`] = `
<Tabs
  destroyInactiveTabPane={true}
  navWrapper={[Function]}
  onChange={[Function]}
  prefixCls="ts-rc-tabs"
  renderTabBar={[Function]}
  renderTabContent={[Function]}
  style={Object {}}
  tabBarPosition="top"
/>
`;
